etc/systemd/dnf-system-upgrade.service

```diff
@@ -1,14 +1,16 @@
 [Unit]
-Description=System Upgrade
+Description=System Upgrade using DNF
 ConditionPathExists=/system-update/.dnf-system-upgrade
 Documentation=http://www.freedesktop.org/wiki/Software/systemd/SystemUpdates
 
 DefaultDependencies=no
 Requires=sysinit.target
 After=sysinit.target systemd-journald.socket
-Before=shutdown.target
+Before=shutdown.target system-update.target
```

Comment:
The ordering dependency with the target is added automatically when DefaultDependencies=yes. But this unit has DefaultDependencies=no, so it's necessary to make any ordering manually.